lib/extensions/attachable/active_record/base.rb

```diff
@@ -27,6 +41,51 @@ def files=(files)
         attachment_references.build(file: file)
       end
     end
+
+    private
+
+    # Delete the attachment references which are not in the content.
+    def update_attachment_references
+      return if attachment_references.empty?
+
+      attachment_reference_ids = active_attachment_reference_ids
```

Comment:
There's a cyclic dependency here. How do you solve it? The attachment needs to be created first otherwise there is nothing to reference.

The current one just retrieves IDs from the content, it does not resolve this problem.